File: apps/mobile/ios/SlateApp/WatchSessionBridge.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(WatchSessionBridge, NSObject)

RCT_EXTERN_METHOD(syncToken:(NSString *)sessionJSON
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(clearToken:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup {
  return NO;
}

@end

File: apps/mobile/ios/SlateApp/WatchSessionBridge.swift
import Foundation

/// React Native bridge module to expose WatchSessionManager to JS
@objc(WatchSessionBridge)
final class WatchSessionBridge: NSObject {

    @objc func syncToken(_ sessionJSON: String,
                         resolve: @escaping RCTPromiseResolveBlock,
                         reject: @escaping RCTPromiseRejectBlock) {
        WatchSessionManager.shared.syncToken(sessionJSON)
        resolve(true)
    }

    @objc func clearToken(_ resolve: @escaping RCTPromiseResolveBlock,
                          reject: @escaping RCTPromiseRejectBlock) {
        WatchSessionManager.shared.clearToken()
        resolve(true)
    }

    @objc static func requiresMainQueueSetup() -> Bool {
        return false
    }
}
